<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyAgreeDlg</class>
 <widget class="QDialog" name="KeyAgreeDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>580</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Key Agreement</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="5" column="0" colspan="2">
    <widget class="Line" name="line_5">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QPlainTextEdit" name="mSecretKeyText"/>
   </item>
   <item row="6" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="mClearDataAllBtn">
       <property name="text">
        <string>Clear Data All</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="mCloseBtn">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QTabWidget" name="mTabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>DH</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="1" column="0" colspan="4">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QLineEdit" name="mBPublicKeyText"/>
       </item>
       <item row="7" column="1">
        <widget class="QLineEdit" name="mBPrivateKeyText"/>
       </item>
       <item row="8" column="2">
        <widget class="QLineEdit" name="mBPublicKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="mAPrivateKeyText"/>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>B PrivateKey</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="4">
        <layout class="QGridLayout" name="gridLayout_10">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="0" column="1">
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Len</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLineEdit" name="mPLenText">
           <property name="minimumSize">
            <size>
             <width>40</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>p value</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="mAPublicKeyText"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>A PublicKey</string>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QPushButton" name="mAGenDHKeyBtn">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>A GenDHKey</string>
         </property>
        </widget>
       </item>
       <item row="8" column="3">
        <widget class="QPushButton" name="mBGenDHKeyBtn">
         <property name="text">
          <string>B GenDHKey</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>B PublicKey</string>
         </property>
        </widget>
       </item>
       <item row="7" column="2">
        <widget class="QLineEdit" name="mBPrivateKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QPushButton" name="mADHPriBtn">
         <property name="text">
          <string>A GenDHPri</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QLineEdit" name="mAPrivateKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="4">
        <layout class="QGridLayout" name="gridLayout_12">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="0" column="4">
          <widget class="QComboBox" name="mGCombo"/>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>g value</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="mLengthText">
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QPushButton" name="mGenParamBtn">
           <property name="text">
            <string>GenParam</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Parameter Length</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <spacer name="horizontalSpacer_7">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="3" column="0" colspan="4">
        <widget class="QPlainTextEdit" name="mPText"/>
       </item>
       <item row="5" column="2">
        <widget class="QLineEdit" name="mAPublicKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="3">
        <widget class="QPushButton" name="mBDHPriBtn">
         <property name="text">
          <string>B GenDHPri</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>A PrivateKey</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="4">
        <widget class="Line" name="line_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>ECDH</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="3" column="2">
        <widget class="QLineEdit" name="mAECDHPubKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>B PublicKey</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0" colspan="3">
        <layout class="QGridLayout" name="gridLayout_9">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="0" column="3">
          <widget class="QPushButton" name="mBGenKeyPairBtn">
           <property name="text">
            <string>GenKeyPair</string>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QPushButton" name="mBFindPriKeyBtn">
           <property name="text">
            <string>FindPriKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QPushButton" name="mBGenPriKeyBtn">
           <property name="text">
            <string>GenPriKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="mBCheckPubKeyBtn">
           <property name="text">
            <string>CheckPubKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QPushButton" name="mBGenPubKeyBtn">
           <property name="text">
            <string>GenPubKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="mBCheckKeyPairBtn">
           <property name="text">
            <string>CheckKeyPair</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="9" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0" colspan="3">
        <layout class="QGridLayout" name="gridLayout_11">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="0" column="6">
          <widget class="QPushButton" name="mAFindPriKeyBtn">
           <property name="text">
            <string>FindPriKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QPushButton" name="mAGenPubKeyBtn">
           <property name="text">
            <string>GenPubKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="mACheckPubKeyBtn">
           <property name="text">
            <string>CheckPubKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QPushButton" name="mAGenPriKeyBtn">
           <property name="text">
            <string>GenPriKey</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="3">
          <widget class="QPushButton" name="mAGenKeyPairBtn">
           <property name="text">
            <string>GenKeyPair</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="mACheckKeyPairBtn">
           <property name="text">
            <string>CheckKeyPair</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="6" column="2">
        <widget class="QLineEdit" name="mBECDHPriKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="3">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>B PrivateKey</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="mAECDHPubKeyText"/>
       </item>
       <item row="0" column="0" colspan="3">
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_12">
           <property name="minimumSize">
            <size>
             <width>80</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Parameter</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="mECDHParamCombo"/>
         </item>
         <item row="0" column="2">
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="6" column="1">
        <widget class="QLineEdit" name="mBECDHPriKeyText"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_6">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>A PrivateKey</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="mAECDHPriKeyText"/>
       </item>
       <item row="7" column="2">
        <widget class="QLineEdit" name="mBECDHPubKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLineEdit" name="mAECDHPriKeyLenText">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>A PublicKey</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QLineEdit" name="mBECDHPubKeyText"/>
       </item>
       <item row="5" column="0" colspan="3">
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="0">
    <layout class="QGridLayout" name="gridLayout_4">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item row="0" column="3">
      <widget class="QLineEdit" name="mSecretKeyLenText">
       <property name="minimumSize">
        <size>
         <width>40</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>40</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Secret Key</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Len</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="0">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item row="0" column="3">
      <widget class="QPushButton" name="mBCalcBtn">
       <property name="text">
        <string>B Calculate</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="mSecretClearBtn">
       <property name="text">
        <string>Secret Clear</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="mACalcBtn">
       <property name="text">
        <string>A Calculate</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <spacer name="horizontalSpacer_8">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>mTabWidget</tabstop>
  <tabstop>mLengthText</tabstop>
  <tabstop>mGCombo</tabstop>
  <tabstop>mGenParamBtn</tabstop>
  <tabstop>mPLenText</tabstop>
  <tabstop>mPText</tabstop>
  <tabstop>mAPrivateKeyText</tabstop>
  <tabstop>mAPrivateKeyLenText</tabstop>
  <tabstop>mADHPriBtn</tabstop>
  <tabstop>mAPublicKeyText</tabstop>
  <tabstop>mAPublicKeyLenText</tabstop>
  <tabstop>mAGenDHKeyBtn</tabstop>
  <tabstop>mBPrivateKeyText</tabstop>
  <tabstop>mBPrivateKeyLenText</tabstop>
  <tabstop>mBDHPriBtn</tabstop>
  <tabstop>mBPublicKeyText</tabstop>
  <tabstop>mBPublicKeyLenText</tabstop>
  <tabstop>mBGenDHKeyBtn</tabstop>
  <tabstop>mECDHParamCombo</tabstop>
  <tabstop>mAECDHPriKeyText</tabstop>
  <tabstop>mAECDHPriKeyLenText</tabstop>
  <tabstop>mAECDHPubKeyText</tabstop>
  <tabstop>mAECDHPubKeyLenText</tabstop>
  <tabstop>mACheckPubKeyBtn</tabstop>
  <tabstop>mACheckKeyPairBtn</tabstop>
  <tabstop>mAGenKeyPairBtn</tabstop>
  <tabstop>mAGenPriKeyBtn</tabstop>
  <tabstop>mAGenPubKeyBtn</tabstop>
  <tabstop>mAFindPriKeyBtn</tabstop>
  <tabstop>mBECDHPriKeyText</tabstop>
  <tabstop>mBECDHPriKeyLenText</tabstop>
  <tabstop>mBECDHPubKeyText</tabstop>
  <tabstop>mBECDHPubKeyLenText</tabstop>
  <tabstop>mBCheckPubKeyBtn</tabstop>
  <tabstop>mBCheckKeyPairBtn</tabstop>
  <tabstop>mBGenKeyPairBtn</tabstop>
  <tabstop>mBGenPriKeyBtn</tabstop>
  <tabstop>mBGenPubKeyBtn</tabstop>
  <tabstop>mBFindPriKeyBtn</tabstop>
  <tabstop>mSecretKeyLenText</tabstop>
  <tabstop>mSecretKeyText</tabstop>
  <tabstop>mClearDataAllBtn</tabstop>
  <tabstop>mCloseBtn</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
